deployment/provision_vagrant_vm.sh

```diff
@@ -36,21 +36,27 @@ service redis-server restart
 # install apache
 apt-get -q install -y apache2 apache2-dev libapache2-mod-wsgi-py3
 
-# With docker for mac, root will own the mount point (uid=0). chmod does not touch the host file system in these cases.
-OWNER=$(stat -c %U "$MOUNTPOINT/evap")
-if [ "$OWNER" == "root" ]; then chown -R 1042 "$MOUNTPOINT"; fi
+# link the mounted evap folder from the home directory:
+OWNER=$(stat -c %u "$MOUNTPOINT/evap")
+if [ "$OWNER" != 1042 ]; then
+  #  if, for any reason, vagrant failed to mount evap under the correct uid, use bindfs to link it to /opt/evap using the wanted user id
+  #  If you are using docker: Make sure /dev/fuse of your host is forwarded into the container
+  apt-get -q install -y bindfs
+  mkdir -p "$REPO_FOLDER"
+  bindfs --map="$OWNER/1042:@$OWNER/@1042" "$MOUNTPOINT" "$REPO_FOLDER" || exit 1
+  echo "sudo bindfs --map=$OWNER/1042:@$OWNER/@1042 '$MOUNTPOINT' '$REPO_FOLDER'" >> /home/$USER/.bashrc
```

Comment:
for docker rootless, `stat -c "%u" /evap/evap` returns 0, so the creation of the user fails directly. Its uid should be 1042 instead

Reply:
@richardebeling I am very confused. Who do you mean with "my user"? `vagrant`, `evap` or your host user?

Reply:
I mean the "evap" user inside docker. In line 53 of the provisioning script, the user is created with the user id that is the owner of `$MOUNTPOINT/evap", which is the vagrant-caused mountpoint of the repo folder, and which is owned by uid 1000 at that point (with docker-ce on ubuntu 22.04 after running `vagrant up --provider=docker`).

Note that there is currently (latest commit in this PR) an inconsistency in the provision script: We have a owner-changing bindfs-mount from `$MOUNTPOINT` to `$REPO_FOLDER` to fix ownership, but the rest of the script works in `$MOUNTPOINT`. Before, this didn't matter as we fixed ownership for both, but now they have different ownership, so working inside `$MOUNTPOINT` doesn't seem reasonable to me. The current effect of this is that the provisioning script finishes without error (since everything is setup for the user with id 1000), but attempting to use the environment after `vagrant ssh` fails since evap with uid 1000 doesn't own any of the 1042-owned files in `/evap/evap`. It would work in `/evap/evap`.